### PR TITLE
Add ConstIsqrt trait (isqrt, checked_isqrt)

### PR DESCRIPTION
**Pull request by kaidokert**
More c0nst

## Summary by Sourcery

Add a const-compatible integer square root API and implement it for primitive unsigned integers and FixedUInt types.

New Features:
- Introduce the ConstIsqrt trait providing isqrt and checked_isqrt for const contexts.
- Implement ConstIsqrt for unsigned primitive integer types via a shared macro.
- Add a ConstIsqrt implementation and const helper for FixedUInt using a bit-by-bit integer square root algorithm.

Tests:
- Add unit tests verifying ConstIsqrt behavior for FixedUInt across perfect squares, non-squares, larger values, checked_isqrt semantics, and general correctness including const evaluation.

## Summary by CodeRabbit

* **New Features**
  * Added compile-time integer square-root for unsigned primitives and fixed-size integers, with const-compatible and safe checked variants.

* **Tests**
  * Added unit tests covering correctness, edge cases, larger values, and const-evaluation paths.

* **Chores / Documentation**
  * Bumped minimum Rust toolchain to 1.73 and updated the README badge accordingly.

### CODE REVIEW

File: src/fixeduint/isqrt_impl.rs [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+                return Some(Self::zero());
+            }
+
+            // Start with the highest bit position that could be set in the result
+            // For sqrt, the result has at most half the bits of the input
+            let mut result = Self::zero();
+
+            // Find starting bit position: half of the bit length of self
+            let bit_len = Self::BIT_SIZE - ConstPrimInt::leading_zeros(self) as usize;
+            let start_bit = bit_len.div_ceil(2);
```
[the remark this answers is not in the document]
Comment (kaidokert) [reply src/fixeduint/isqrt_impl.rs]: I'm going to feature-gate these things on MSRV. For now can you just add a follow-up ticket to add MSRV testing properly ? I think in trunk i have other msrv > 1.51 violations elsewhere, so these all need to be gated
